RELEASE.md

```diff
@@ -0,0 +1,25 @@
+# Doing a release
+
+## Tag the release
+
+1. Pick new version V.
+1. Make sure [release notes](docs/release_notes.rst) on main branch have latest changelog entries with version V.
+1. In GitHub UI, added release (or tag) with name `v<V>`, e.g. `v1.17.23`.
+
+## Upload
+
+At this point you build wheels and sdist:
+
+```
+$ python setup.py sdist
+$ pip wheel -w dist/ --no-deps .
```

Comment:
After supporting automatic publishing, we should no longer need this step. But anyway, [build](https://pypi.org/project/build) is the preferred way to build distributions now.